File: Exemplos/App_eXperience_Flutter/E1_Flutter-main/lib/services/services_tef/sitefReturn.dart
class SitefReturn {
  String cODAUTORIZACAO = '';
  String vIAESTABELECIMENTO = '';
  String cOMPDADOSCONF = '';
  String bANDEIRA = '';
  String nUMPARC = '';
  String cODTRANS = '';
  String rEDEAUT = '';
  String nSUSITEF = '';
  String vIACLIENTE = '';
  String vLTROCO = '';
  String tIPOPARC = '';
  String cODRESP = '';
  String nSUHOST = '';

  SitefReturn({
    this.cODAUTORIZACAO = '',
    this.vIAESTABELECIMENTO = '',
    this.cOMPDADOSCONF = '',
    this.bANDEIRA = '',
    this.nUMPARC = '',
    this.cODTRANS = '',
    this.rEDEAUT = '',
    this.nSUSITEF = '',
    this.vIACLIENTE = '',
    this.vLTROCO = '',
    this.tIPOPARC = '',
    this.cODRESP = '',
    this.nSUHOST = '',
  });

  SitefReturn.fromJson(Map<String, dynamic> json) {
    cODAUTORIZACAO = json['COD_AUTORIZACAO'] ?? '';
    vIAESTABELECIMENTO = json['VIA_ESTABELECIMENTO'] ?? '';
    cOMPDADOSCONF = json['COMP_DADOS_CONF'] ?? '';
    bANDEIRA = json['BANDEIRA'] ?? '';
    nUMPARC = json['NUM_PARC'] ?? '';
    cODTRANS = json['CODTRANS'] ?? '';
    rEDEAUT = json['REDE_AUT'] ?? '';
    nSUSITEF = json['NSU_SITEF'] ?? '';
    vIACLIENTE = json['VIA_CLIENTE'] ?? '';
    vLTROCO = json['VLTROCO'] ?? '';
    tIPOPARC = json['TIPO_PARC'] ?? '';
    cODRESP = json['CODRESP'] ?? '';
    nSUHOST = json['NSU_HOST'] ?? '';
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['COD_AUTORIZACAO'] = this.cODAUTORIZACAO;
    data['VIA_ESTABELECIMENTO'] = this.vIAESTABELECIMENTO;
    data['COMP_DADOS_CONF'] = this.cOMPDADOSCONF;
    data['BANDEIRA'] = this.bANDEIRA;
    data['NUM_PARC'] = this.nUMPARC;
    data['CODTRANS'] = this.cODTRANS;
    data['REDE_AUT'] = this.rEDEAUT;
    data['NSU_SITEF'] = this.nSUSITEF;
    data['VIA_CLIENTE'] = this.vIACLIENTE;
    data['VLTROCO'] = this.vLTROCO;
    data['TIPO_PARC'] = this.tIPOPARC;
    data['CODRESP'] = this.cODRESP;
    data['NSU_HOST'] = this.nSUHOST;
    return data;
  }
}

File: Exemplos/App_eXperience_ReactNative/Elgin_AppExperience_ReactNative-font/src/services/services_tef/msitef/sitefReturn.js
export default class SitefReturn {
  convertJSONtoMSITEF(json) {
    this.cODAUTORIZACAO = json.COD_AUTORIZACAO;
    this.viaESTABELECIMENTO = json.VIA_ESTABELECIMENTO;
    this.cOMPDADOSCONF = json.COMP_DADOS_CONF;
    this.bANDEIRA = json.BANDEIRA;
    this.numPARC = json.NUM_PARC;
    this.cODTRANS = json.CODTRANS;
    this.rEDEAUT = json.REDE_AUT;
    this.nSUSITEF = json.NSU_SITEF;
    this.viaCLIENTE = json.VIA_CLIENTE;
    this.vLTROCO = json.VLTROCO;
    this.tipoPARC = json.TIPO_PARC;
    this.cODRESP = json.CODRESP;
    this.nSUHOST = json.NSU_HOST;

    return json;
  }

  receiveResultInJSON(resultInJSON) {
    resultInJSON = resultInJSON.toString().split('\\r').join('');
    resultInJSON = resultInJSON.toString().split('"{').join('{');
    resultInJSON = resultInJSON.toString().split('}"').join('}');

    var resultParsed = JSON.parse(resultInJSON);

    this.convertJSONtoMSITEF(resultParsed);
  }

  vIAESTABELECIMENTO() {
    return this.viaESTABELECIMENTO ? this.viaESTABELECIMENTO : '';
  }

  setCodTrans(cODTRANS) {
    this.cODTRANS = cODTRANS;
  }

  getcODAUTORIZACAO() {
    return this.cODAUTORIZACAO ? this.cODAUTORIZACAO : '';
  }
  cOMPDADOSCONF() {
    return this.cOMPDADOSCONF ? this.cOMPDADOSCONF : '';
  }
  bANDEIRA() {
    return this.bANDEIRA ? this.bANDEIRA : '';
  }
  nUMPARC() {
    if (this.numPARC != null) {
      return this.numPARC;
    }
    return '';
  }
  cODTRANS() {
    return this.cODTRANS ? this.cODTRANS : '';
  }
  rEDEAUT() {
    return this.rEDEAUT ? this.rEDEAUT : '';
  }
  nSUSITEF() {
    return this.nSUSITEF;
  }
  vIACLIENTE() {
    return this.viaCLIENTE ? this.viaCLIENTE : '';
  }
  vLTROCO() {
    return this.vLTROCO;
  }
  tipoPARC() {
    return this.tipoPARC;
  }
  getcODRESP() {
    return this.cODRESP ? this.cODRESP : '';
  }
  nSUHOST() {
    return this.nSUHOST;
  }

  getNameTransCod() {
    var retorno = 'Valor invalido';

    switch (parseInt(this.tipoPARC, 10)) {
      case 0:
        retorno = 'A vista';
        break;
      case 1:
        retorno = 'Pré-Datado';
        break;
      case 2:
        retorno = 'Parcelado Loja';
        break;
      case 3:
        retorno = 'Parcelado Adm';
        break;
    }

    return retorno;
  }
}
